### PR TITLE
Tests updated to NSubstitute 4

### PR DESCRIPTION
**Pull request by kanadaj**
All tests now work with NSubstitute 4.2.2 instead of 3.x. 

`LogDebug_DefaultOptions_DoesNotRecordsBreadcrumbs` and `LogTrace_DefaultOptions_DoesNotRecordsBreadcrumbs` have had their `Args.Any<>()` replaced with `DidNotReceiveWithAnyArgs()` since this is a stricter condition anyway and the old version doesn't seem to work regardless of what I've tried.

#skip-changelog

### CODE REVIEW

**Comment by kanadaj** [comment]
Addresses https://github.com/getsentry/sentry-dotnet/issues/626

File: test/Sentry.Extensions.Logging.Tests/SentryLoggerTests.cs [hunk 1]
```diff
@@ -370,14 +370,25 @@ public void LogDebug_DefaultOptions_DoesNotRecordsBreadcrumbs()
...
+
+            // The below test is the original one, but for some reason NSubstitute 4 refuses to use the Arg.Any() parameters even though they should be correct and therefore throws an error
+            // _fixture.Hub.DidNotReceive()
+            //     .AddBreadcrumb(
+            //         _fixture.Clock,
+            //         Arg.Any<string>(),
+            //         _fixture.CategoryName,
+            //         BreadcrumbType,
+            //         null,
+            //         Arg.Any<BreadcrumbLevel>());
```
Comment (bruno-garcia) [inline test/Sentry.Extensions.Logging.Tests/SentryLoggerTests.cs:391]: Makes sense
```suggestion
```

File: test/Sentry.Extensions.Logging.Tests/SentryLoggerTests.cs [hunk 2]
```diff
@@ -387,14 +398,24 @@ public void LogTrace_DefaultOptions_DoesNotRecordsBreadcrumbs()
...
```
Comment (bruno-garcia) [inline test/Sentry.Extensions.Logging.Tests/SentryLoggerTests.cs:418]:
```suggestion
```

**Review by bruno-garcia — APPROVED** [review]
Nice, that's all? :) I was under the impressino would be a lot more code.

Please clean up the dead code and we're good to go

**Pull request merged**